static/js/webrtc.js

```diff
@@ -234,52 +234,79 @@ var rtc = (function() {
           videoContainer.addClass('local-user');
           video.muted = true;
         }
-        self.addInterface(userId);
+        self.addInterface(userId, stream);
       }
       if (stream) {
         attachMediaStream(video, stream);
       } else if (video) {
         $(video).parent().remove();
       }
     },
-    addInterface: function(userId) {
+    addInterface: function(userId, stream) {
       var isLocal = userId == self.getUserId();
       var videoId = "video_" + userId.replace(/\./g, "_");
       var $video = $("#" + videoId);
 
+      var audioTrack = stream.getAudioTracks()[0];
+      var initiallyMuted = false;
+      if (audioTrack) {
+        initiallyMuted = !audioTrack.enabled
+      }
+
       var $mute = $("<span class='interface-btn audio-btn buttonicon'>")
-        .attr("title", "Mute")
-        .on({
-          click: function(event) {
-            var muted;
-            if (isLocal) {
-              muted = self.toggleMuted();
-            } else {
-              $video[0].muted = !$video[0].muted;
-              muted = $video[0].muted;
-            }
-            $mute
-              .attr("title", muted ? "Unmute" : "Mute")
-              .toggleClass("muted", muted);
+        .attr("title", clientVars.webrtc.audio_enabled
+          ? (initiallyMuted ? "Unmute" : "Mute")
+          : "Audio disabled by admin")
+        .toggleClass("muted", initiallyMuted)
+
+      // TODO - is audio_enabled/video_enabled a security issue? Like do we care if the user hacks the client to do video anyway?
+      // I suppose this has nothing to do with the server anyway. It uses Google turn servers etc
```

Comment:
@JohnMcLear If I understand webrtc correctly, it doesn't touch the etherpad-lite server at all, right? Just these stun and turn servers which are usually run by somebody else?

I only ask to make sure we're not missing any security requirements server-side. But if I understand webrtc right, I think that there's nothing we can do to stop some tricky user from turning on audio and video chat despite the site admin's wishes.